src/Dapr.Actors/Runtime/DaprStateProvider.cs

```diff
@@ -19,10 +19,12 @@ namespace Dapr.Actors.Runtime
     internal class DaprStateProvider
     {
         private readonly IActorStateSerializer actorStateSerializer;
+        private readonly JsonSerializerOptions jsonSerializerOptions;
 
-        public DaprStateProvider(IActorStateSerializer actorStateSerializer = null)
+        public DaprStateProvider(IActorStateSerializer actorStateSerializer = null, JsonSerializerOptions jsonSerializerOptions = null)
```

Comment:
It seems like this should really be two separate constructors right?

If you specify a serializer, then the json options have no effect. The bulk of the logic here could be in an implementation of `IActorStateSerializer`- with a constructor here that accepts the options and instantiates it.